### ISSUE TITLE
[Question] Plan to port flake into to nixpkgs

### ISSUE THREAD

**Issue by amerinor01** [nilp0inter/autofirma-nix#20]
Hello, @nilp0inter. First, thank you for making autofirma available in the Nix ecosystem. I plan to port this flake into nixpkgs, at least the `autofirma` for now. But I have some questions.

*  do you know if the `pr-367.patch` will be merged into the application soon?
* Shouldn't the `detect_java_version.patch` patch also be pushed to the official repository?

**Comment by nilp0inter** [comment]
Hello, @amerinor01. Thank you for your interest in porting autofirma-nix to nixpkgs and for your kind words.

To address your questions:

1. While the project's documentation does not explicitly mention integration into nixpkgs, it is indeed part of our roadmap. However, we need to stabilize the project and resolve some outstanding issues before proceeding (details below).
2. In our experience, AutoFirma is a government software project that typically does not prioritize pull requests, especially those that do not impact officially supported operating systems (Windows, Mac, Debian, and Fedora). Therefore, it is unlikely that [pr-367](https://github.com/ctt-gob-es/clienteafirma/pull/367) will be merged anytime soon.
3. The detect-java-version.patch falls under the same category. I haven't submitted a pull request for it due to similar concerns, but it might be worth considering.

Before we can integrate this into nixpkgs, we need to address a few issues:

- [x] **Robust Compilation Process**: We have recently made the compilation process more robust. The previous brittle modifications to pom.xml are now handled using xmlstarlet, enhancing stability.
- [ ] **Certificate Rejection Issue**: AutoFirma currently rejects some certificates from legitimate sites. This might be related to how NixOS manages the Java truststore, but further investigation is needed. Some efforts have been made in this area, which you can check out [here](https://discourse.nixos.org/t/custom-ssl-certificates-for-jdk/18297/11?u=nilp0inter).
- [ ] **Adoption of `maven.buildMavenPackage`**: Now that the compilation process is more stable, we should consider using the official `maven.buildMavenPackage`, which would be beneficial if we integrate into nixpkgs.

Your efforts and contributions are much appreciated. If you have any further questions or need assistance, please feel free to reach out.

Closing now, but feel free to add more comments if needed.

Best regards,  
@nilp0inter
